metrics/registry.go

```diff
@@ -0,0 +1,138 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+func NewNamespaceRegistry(n string) *NamespaceRegistry {
+	return &NamespaceRegistry{
+		Namespace: n,
+		mr:        NewMetricDescRegistry(),
+		m:         &sync.Map{},
+	}
+}
+
+type NamespaceRegistry struct {
+	Namespace string
+
+	mr *MetricDescRegistry
+
+	// Map of collectable -> *MetricDescRegistry
+	m *sync.Map
+}
+
+func (n *NamespaceRegistry) Describe(c chan<- MetricDesc) error {
+	c <- MetricDesc{
+		Name:   n.Namespace,
+		Prefix: true,
```

Comment:
I didn't notice this being used; I may have missed it.

Reply:
This makes a registry implement a Collectable. Its used if/when you register a registry to a registry.